File: Training/Classic Puzzle - Easy/Ghost_legs.cpp
/**
 * @file Ghost_Legs.cpp
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-10-11
 * 
 * @copyright Copyright (c) 2021
 */

#include <iostream>
#include <vector>

using namespace std;

int main(){
    int W, H;
    cin >> W >> H; cin.ignore();

    vector<vector<char>> myMatrix;
    for (int i = 0; i < H; i++) {
        string line;
        getline(cin, line);

        vector<char> tempLine;
        for (auto const &c: line)
            tempLine.push_back(c);
        myMatrix.push_back(tempLine);
    }

    for (size_t j = 0; j < W ; j+=3){
        size_t actualCol = j;
        size_t actualLine = 0;

        while (actualLine != H - 1){
            
            if(actualCol >= 3 && myMatrix[actualLine][actualCol-1] == '-')
                actualCol -= 3;

            else if(actualCol <= W-3 && myMatrix[actualLine][actualCol+1] == '-')
                actualCol += 3;

            actualLine += 1;
        }

        cout << myMatrix[0][j] << myMatrix[actualLine][actualCol] << endl;
    }
}

File: Training/Classic Puzzle - Easy/Ghost_Legs.py
"""
 @file Ghost_Legs.py
 @author Ladjouzi Rachid (RashidLadj@gmail.com)
 @brief 
 @version 0.1
 @date 2021-10-12
 
 @copyright Copyright (c) 2021
"""

w, h = [int(i) for i in input().split()]
myMatrix= []
for i in range(h):
    myMatrix.append(input())

for j in range (0, w, 3):
    actualCol = j
    actualLine = 0
    
    while actualLine != h - 1:   
        if actualCol >= 3 and myMatrix[actualLine][actualCol-1] == '-' :
            actualCol -= 3

        elif actualCol <= w-3 and myMatrix[actualLine][actualCol+1] == '-':
            actualCol += 3

        actualLine += 1


    print(myMatrix[0][j]+myMatrix[actualLine][actualCol])